Workbooks/Azure Monitor - Workspaces/Container Insights Usage/Container Insights Usage.workbook

```diff
@@ -0,0 +1,257 @@
+      },
+      "name": "text - 2",
+      "styleSettings": {
+        "progressStyle": "loader",
+        "showBorder": true
+      }
+    },
+    {
+      "type": 9,
+      "content": {
+        "version": "KqlParameterItem/1.0",
+        "parameters": [
+          {
+            "id": "cbafb196-4c70-4f43-b189-1de7620db19e",
+            "version": "KqlParameterItem/1.0",
+            "name": "Time_range",
+            "type": 4,
+            "description": "Select time-range for data selection",
+            "isRequired": true,
+            "value": {
+              "durationMs": 7776000000
+            },
+            "typeSettings": {
+              "selectableValues": [
+                {
+                  "durationMs": 300000
+                },
+                {
+                  "durationMs": 900000
+                },
+                {
+                  "durationMs": 1800000
+                },
+                {
+                  "durationMs": 14400000
+                },
+                {
+                  "durationMs": 43200000
+                },
+                {
+                  "durationMs": 86400000
+                },
+                {
+                  "durationMs": 172800000
+                },
+                {
+                  "durationMs": 259200000
+                },
+                {
+                  "durationMs": 604800000
+                },
+                {
+                  "durationMs": 1209600000
+                },
+                {
+                  "durationMs": 2419200000
+                },
+                {
+                  "durationMs": 2592000000
+                },
+                {
+                  "durationMs": 5184000000
+                },
+                {
+                  "durationMs": 7776000000
+                }
+              ]
+            },
+            "label": "Time Range"
+          }
+        ],
+        "style": "pills",
+        "queryType": 0,
+        "resourceType": "microsoft.operationalinsights/workspaces"
+      },
+      "name": "parameters - 6"
+    },
+    {
+      "type": 3,
+      "content": {
+        "version": "KqlItem/1.0",
+        "query": "Usage\r\n| where TimeGenerated {Time_range}\r\n| where IsBillable == true\r\n| summarize TotalVolumeGB = sum(Quantity) / 1000 by bin(TimeGenerated, 1d), Solution\r\n| render linechart",
```

Comment:
instead of hardcoding the grain to always be 1d in the charts no matter the time range, you can do `bin(TimeGenerated, {Time_range:grain})` to get an automatically set grain depending on the time range.  at 90d it would still be 1d, but at 1d selected for time range, you'd get 30m grains, etc.

Reply:
Fixed